src/chainparams.cpp

```diff
@@ -242,6 +244,7 @@ class CRegTestParams : public CChainParams {
         vSeeds.clear();  //! Regtest mode doesn't have any DNS seeds.
 
         fMiningRequiresPeers = false;
+        fMiningRequiresConfiguration = false;
```

Comment:
Please remove those. Each difference between regnet and mainnet makes our tests a bit less useful.

You can adjust this in `qa/rpc-tests/test_framework/util.py` or even better in the respective `qa/rpc-tests` that cover  this.